plugins/ccgx/fu-ccgx-cyacd-file.c

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright (C) 2020 Cypress Semiconductor Corporation.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-ccgx-cyacd-file.h"
+
+static guint8
+ch_to_hex (guint8 value)
+{
+	if ('0' <= value && value <= '9') {
+		return (guint8)(value - '0');
+	}
+
+	if ('a' <= value && value <= 'f') {
+		return (guint8)(10 + value - 'a');
+	}
+
+	if ('A' <= value && value <= 'F') {
+		return (guint8)(10 + value - 'A');
+	}
+	return 0;
+}
+
+static void
+convert_ascii_to_hex (guint8 *src_buffer, guint32 src_size, guint8 *dst_buffer, guint32 dst_size)
+{
+	guint32 idx;
+	guint32 src_index = 0;
+
+	g_return_if_fail (src_buffer != NULL && src_size > 0);
+	g_return_if_fail (dst_buffer != NULL && dst_size > 0);
+
+	for (idx = 0; idx < dst_size; idx++) {
+		src_index = 2 * idx;
+		if (src_index + 1 >= src_size) {
+			break;
+		}
+		dst_buffer[idx] = (ch_to_hex (src_buffer [src_index]) << 4) | ch_to_hex (src_buffer [src_index + 1]);
+	}
+}
+
+static gboolean
+cyacd_buffer_trim_cr_lf (CyacdFileHandle *handle)
+{
+	guint8 ch;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+
+	while (handle->pos < handle->buffer_size) {
+		ch = handle->buffer [handle->pos];
+		if (ch != '\n' &&  ch != '\r') {
+			return TRUE;
+		}
+		handle->pos++;
+	}
+	return	FALSE;
+}
+
+static guint32
+cyacd_buffer_read_line (CyacdFileHandle* handle, guint8 *line_buffer, guint32 line_buff_size)
+{
+	guint32 index = 0;
+	guint8 ch;
+
+	cyacd_buffer_trim_cr_lf (handle);
+
+	while (handle->pos < handle->buffer_size && index < line_buff_size) {
+		ch = handle->buffer [handle->pos];
+		handle->pos++;
+
+		if (ch == '\n' || ch == '\r') {
+			break;
+		}
+
+		line_buffer [index] = ch;
+		index++;
+	}
+
+	cyacd_buffer_trim_cr_lf (handle);
+	return index;
+}
+
+/**
+ * fu_ccgx_cyacd_file_init_handle:
+ * @handle_array[in,out] Cyacd File Handle Array
+ * @num_of_array number of Cyand File Handle Arrray
+ * @buffer cyacd data buffer
+ * @buffer_size cyacd data buffer size
+ *
+ * Initialize and setup up cyacd handle with user buffer
+ *
+ * Returns: number of valid handle
+*/
+
```

Comment:
as a nit, any of these documentation comments should be directly above the function rather than a newline between tehm

Reply:
I will remove a unnecessary new line.

Reply:
I fixed it.